detection_rules/kbwrap.py

```diff
@@ -67,7 +70,36 @@ def upload_rule(ctx, toml_files):
         payload = rule.contents.copy()
         meta = payload.setdefault("meta", {})
         meta["original"] = dict(id=rule.id, **rule.metadata)
-        payload["rule_id"] = str(uuid4())
+
+        # Determine the rule id to use.
+        # Github issue 612
+        if preserve_id and not force:
+            with kibana:
+                # determine if the rule already exists
+                try:
+                    resp = kibana.request(
+                        method="GET",
+                        uri="/api/detection_engine/rules",
+                        params={
+                            "rule_id": "{}".format(rule.id)
```

Comment:
```suggestion
                            "rule_id": rule.id
```

Reply:
also, can you use `RuleResource.find_elastic(rule_id=rule.id)` here instead of the `kibana.request()` call?
I think some of this functionality is already encapsulated there, so you shouldn't have too do manual requests here

Reply:
Been trying to get the `RuleResource.find_elastic()` to work, but can't figure it out.
I do get a ResourceIterator, but am unable to determine if the rule then exists and potentially retrieve the name of the rule.

Any idea?